pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -265,6 +288,58 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	return nil
 }
 
+func (c *Reconciler) reconcileMetricsService(ctx context.Context, pa *pav1alpha1.PodAutoscaler) error {
+	logger := logging.FromContext(ctx)
+
+	scale, err := c.kpaScaler.GetScale(ctx, pa)
+	if err != nil {
+		logger.Errorw("Error Getting scale", zap.Error(err))
+		return err
+	}
+	selector, err := labels.ConvertSelectorToLabelsMap(scale.Status.Selector)
+	if err != nil {
+		logger.Errorw(fmt.Sprintf("Error parsing selector string %q", scale.Status.Selector), zap.Error(err))
+		return err
+	}
+	logger.Infof("### PA %s selector: %v", pa.Name, selector)
+
+	sn := resources.MetricsServiceName(pa)
+	// svc, err := c.serviceLister.Services(pa.Namespace).Get(sn)
```

Comment:
Remove please.

Reply:
Actually this is how we should be getting and the line below should go away.

Reply:
It makes tests fail :| Since listers don't update for some reason. Once I find out how to fix it, I'll remove the direct getter.